### PR TITLE
Add support for pip==22.1

### PR DESCRIPTION
**Pull request by atugushev**
- Prepare pip-tools for the upcoming `pip-22.1` release, see https://github.com/pypa/pip/issues/10991.
- Add `main` to the matrix so that we could see the changes pass tests. There is [fail-fast: false](https://github.com/jazzband/pip-tools/blob/ba022e8a6aed95abdf991fac24cd954bd85ac4c6/.github/workflows/ci.yml#L18) on CI so the workflow shouldn't be failed if `main` fails. It would be nice to have some kind of `allow-failure`, but it's not yet implemented, see: https://github.com/actions/toolkit/issues/399.

##### Contributor checklist

- [x] Provided the tests for the changes.
- [x] Assure PR title is short, clear, and good to be included in the user-oriented changelog

##### Maintainer checklist

- [x] Assure one of these labels is present: `backwards incompatible`, `feature`, `enhancement`, `deprecation`, `bug`, `dependency`, `docs` or `skip-changelog` as they determine changelog listing.
- [x] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

File: tox.ini
Comment (pradyunsg) [inline tox.ini:21]:
```suggestion
    piplatest: python -m pip install "pip @ git+https://github.com/pypa/pip.git"
```

**Review by pradyunsg — COMMENTED** [review]
Hurrah! Thanks for picking this up. I have a *very* minor suggestion.

File: tox.ini
Comment (atugushev) [reply tox.ini]: Hey! Thanks for chiming in ^___^ I've reverted my test changes, so this is outdated now.

**Review by ssbarnea — APPROVED** [review]

**Comment by ssbarnea** [comment]
I requests @pradyunsg to update its review so I will wait for his comment or approval before merging. That is a very welcomed addition.

**Comment by pradyunsg** [comment]
Please feel free to merge this without waiting on my review. If I have time to do a review, I’ll do one. :)

**Pull request merged**
